### PR TITLE
refactor(osp): use new company role check

### PR DESCRIPTION
**Pull request by manojava-gk**
## Description

remove customised logic to check company role info in osp page and header component. use company role check in config file to allow user to navigate to osp page.

### Chnagelog

```
**OSP**
 - removed customised logic to check company role info in osp page and header component. use company role check in config file to allow user to navigate to osp page.[#1390](https://github.com/eclipse-tractusx/portal-frontend/pull/1390)
```

## Why

remove custom check for company roles

## Issue

#1108 

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/admin/Dev%20Process/How%20to%20contribute.md#commits-branches-and-pull-requests-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

**Comment by evegufy** [comment]
PR related to https://github.com/eclipse-tractusx/portal-frontend/issues/1328

File: src/components/shared/frame/Header/index.tsx [hunk 1]
```diff
@@ -88,22 +86,6 @@ export const Header = ({
     )
   }, [companyData, companyDetails])
 
-  useEffect(() => {
-    if (
-      companyDetails?.companyRole.includes(
-        CompanyRoleEnum.ONBOARDING_SERVICE_PROVIDER
```
Comment (lavanya-bmw) [inline src/components/shared/frame/Header/index.tsx:94]: Please also remove ONBOARDING_SERVICE_PROVIDER from CompanyRoleEnum  from userApiSlice.ts

**Review by lavanya-bmw — CHANGES_REQUESTED** [review]
Please check or delete options that are not relevant from PR checklist

File: src/components/shared/frame/Header/index.tsx (hunk 1 above)
Comment (manojava-gk) [reply src/components/shared/frame/Header/index.tsx:94]: why? This is being used in ConnectorTypeSelection.tsx

Comment (lavanya-bmw) [reply src/components/shared/frame/Header/index.tsx:94]: Please use COMPANY_ROLES enum from constants.ts file and remove CompanyRoleEnum(from userApiSlice.ts) and its usage as it is a duplicate enum

Comment (manojava-gk) [reply src/components/shared/frame/Header/index.tsx:94]: done

**Review by lavanya-bmw — APPROVED** [review]
lgtm

**Review by kunalgaurav-bmw — APPROVED** [review]

**Review by evegufy — APPROVED** [review]

**Pull request merged**